lib/puppet/provider/scheduled_task/taskscheduler_api2.rb

```diff
@@ -12,8 +13,16 @@
   defaultfor :operatingsystem => :windows
   confine    :operatingsystem => :windows
 
+  def task_scheduler_helper
+    resource[:compatibility] == 1 ? PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2V1Task : PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2Task
+  end
+
   def self.instances
-    PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2V1Task.new.tasks.collect do |job_file|
+    # Works fine
+    # PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2Task.new.tasks.collect do |job_file|
+    # Explodes for some reason, see win32_taskscheduler_spec.rb:696
+    # Something in that test doesn't like the enum for the v2 helper but is fine with the v1 adapter
+    PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2Task.new.tasks.collect do |job_file|
```

Comment:
Getting this to work properly is the last major hurdle to implementation of the wiring (tests probably still needed elsewhere) - as is in master, this will only return legacy jobs. For some reason, pointing to the jobs collection from the new helper blows up. Still investigating but expect it to have something to do with the legacy adapter appending '.job' to the end of job names and the V2 helper not bothering to do so.

Reply:
Yeah, think it was just the tests.

Still need to do some more verification / validation of the PR at https://github.com/puppetlabs/puppetlabs-scheduled_task/pull/27, but at least specs are passing